### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by alysivji**
For some reason flake8 & pytest aren't in the $PATH and the respective make functions fail.

I threw them into requirements, is it because pip -user?

### CODE REVIEW

**Comment by alysivji** [comment]
Also, I vote squash and commit to keep a clean git history.

File: docker-compose.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ services:
        context: .
        dockerfile: Dockerfile
      command: /app/docker-utils/run.sh
+     container_name: myapp
```
Comment (JoeJasinski) [inline docker-compose.yml]: Just a note.  I used to explicitly name my containers, until I found it made it difficult to spin up two instances of the codebase on the same machine.  In practicality, not something that happens too often, but found that the default names that docker picks are decent.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	docker-compose exec app pytest --cov --cov-report html && open ./htmlcov/index.html
+
+test_fast: ## Can pass in parameters using p=''
+	docker-compose exec app pytest $(p)
+
+# Flake 8
+# options: http://flake8.pycqa.org/en/latest/user/options.html
+# codes: http://flake8.pycqa.org/en/latest/user/error-codes.html
+lint: up
+	docker-compose exec app flake8
```
Comment (JoeJasinski) [inline Makefile]: Thank you for making this makefile!

File: docker-compose.yml (hunk 1 above)
Comment (alysivji) [reply docker-compose.yml]: Good call, is there is a way we can find the running container in the current folder?

I can update the `Makefile`

Comment (JoeJasinski) [reply docker-compose.yml]: Ah, I see, you were using the name in the Makefile.  I think attach is the only one were we need the container name. (logs and exec have a docker-compose equivalent form). 

I know you can get the container name with: "docker-compose ps app", but that also gives a bunch of useless header info.  I wonder if the --format  command can be used here... I'm still looking.

Comment (JoeJasinski) [reply docker-compose.yml]: ohh, interesting... I just found this.  I looks like it gets the container hash, which might be useful for the attach command:

```
docker-compose ps -q app
```

File: Makefile (hunk 2 above)
Comment (alysivji) [reply Makefile]: 🙌

File: docker-compose.yml (hunk 1 above)
Comment (alysivji) [reply docker-compose.yml]: Nice! Will update PR in a bit.

**Comment by alysivji** [comment]
I didn't realize I made this in the main repo.

Will be forking and working out of my personal fork + sending PRs back upstream going forward.

**Pull request merged**